runtime/lua/vim/lsp/completion.lua

```diff
@@ -523,6 +705,14 @@ local function trigger(bufnr, clients, ctx)
     end
     local start_col = (server_start_boundary or word_boundary) + 1
     Context.cursor = { cursor_row, start_col }
+    if #matches > 0 and vim.o.completeopt:find('popup') then
+      local group = get_augroup(bufnr)
+      if
+        #api.nvim_get_autocmds({ buffer = bufnr, event = 'CompleteChanged', group = group }) == 0
+      then
+        on_completechanged(group, bufnr)
```

Comment:
defining the autocmd in the middle of trigger() seems strange. why not do it in completion.enable() or some other "one time" setup step?

Reply:
https://github.com/neovim/neovim/pull/32820?notification_referrer_id=NT_kwDOAnvbz7QxNTIxNDEyODY5MTo0MTY3MTYzMQ#discussion_r2040995220

Reply:
> `_omnifunc` does not calling `enable_completion`.

Then `_omnifunc` should call `on_completechanged()` to do the setup. Adding it randomly in `trigger()` is messy.

Reply:
if no matches still call `on_completechanged` ??

Reply:
https://github.com/neovim/neovim/pull/32820#discussion_r2041020778